src/modules/amqpPublisher/AMQPPublisherModule.ts

```diff
@@ -0,0 +1,61 @@
+import { SubscriptionTarget } from "@js-soft/ts-utils";
+import { DataEvent, Event } from "@nmshd/runtime";
+import amqp from "amqplib";
+import { ConnectorRuntimeModule, ConnectorRuntimeModuleConfiguration } from "../../ConnectorRuntimeModule";
+
+export interface AMQPPublisherModuleConfiguration extends ConnectorRuntimeModuleConfiguration {
+    url?: string;
+    exchange?: {
+        name: string;
+        type?: "direct" | "topic" | "headers" | "fanout" | "match";
```

Comment:
Ist es sinnvoll, den Exchange Type in der Config anzugeben? Anders gefragt: ist es relevant für das Modul, welcher Typ es ist? Eigentlich nicht, oder? Ich würde sogar sagen, dass man gefälligst dafür sorgen sollte, dass der Exchange da ist, und nicht die Verantwortung dem Modul überträgt, diesen anzulegen. Oder ist das so üblich? Und ich meine nicht üblich in den 0815 Beispielen, sondern in der "echten" Welt.

Reply:
Ich habe auch schon überlegt ob das sinnvoll ist, aber in den meisten Tutorials wird gesagt, dass man assertExchange aufrufen soll bevor man auf den channel zugreift. Ich könnte allerdings auch checkExchange nutzen und das modul nicht starten lassen wenn der exchange nicht existiert.

Reply:
würde natürlich das modul start verinfachen, wenn man einfach nur optional den namen des exchanges übergibt und der dann schon da sein musst.

Reply:
> Ich könnte allerdings auch checkExchange nutzen und das modul nicht starten lassen wenn der exchange nicht existiert.

Joa, ich glaub so hätte ich es auch gemacht. Und ich hätte dann wahrscheinlich in der Konfiguration auch `exchange: string` gemacht, anstatt ein Objekt.

Reply:
Ja, das meinte ich damit 😆